internal/users/manager.go

```diff
@@ -415,13 +418,24 @@ func getActiveUsers(procDir string) (activeUsers map[string]struct{}, err error)
 	return activeUsers, nil
 }
 
-// GenerateID generates an integer number based on the provided string lowercased.
+// GenerateID deterministically generates an ID between from the given string, ignoring case. The ID is in the range
+// 60000 (the default value of UID_MAX, i.e. the maximum UID for regular users) and MaxInt32.
 func GenerateID(str string) int {
-	lowerCased := strings.ToLower(str)
-	var sum int
-	for i, c := range lowerCased {
-		// Multiplies the value of the rune (which is >0) by its index+1. Subtracts the index to add another layer of conflict prevention.
-		sum += int(c)*(i+1) - i
-	}
-	return (sum % (100000 - 65537)) + 65536 // Ensures that ID is between 65536 and 100000
+	const minID = 60000
+	const maxID = math.MaxInt32 // use MaxInt32 for 32-bit systems
+
+	str = strings.ToLower(str)
+
+	// Create a SHA-256 hash of the input string
+	hash := sha256.Sum256([]byte(str))
+
+	// Convert the first 4 bytes of the hash into an integer
+	number := binary.BigEndian.Uint32(hash[:4])
```

Comment:
We're reducing number to a range smaller than 32 bits, so using more than 32 bits originally will not decrease the probability of collisions.

Reply:
ack, it’s just I don’t know the distribution of the first 4 bytes of sha256 hashes, like, are they evenly distributed? Happy to trust you on that one. As you said, because of modulo, we do have a higher probability of collision on the lower range (which is also even what happened in the previous implementation too, making it worse with the smaller range).

As you wish, if you want to come up with another algorithm, feel free. We can be iterative on this though and consider it’s an improvement and we will refine later on.

Reply:
> it’s just I don’t know the distribution of the first 4 bytes of sha256 hashes, like, are they evenly distributed?

yes, they are

> if you want to come up with another algorithm, feel free

I pushed a new version, that one should generate uniformly distributed IDs

Reply:
yes, looks better, thanks!